xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/test/java/org/xwiki/web/MacrosTest.java

```diff
@@ -287,4 +297,78 @@ void displayUser() throws Exception
             + "<a class=\"user-name\" href=\"/xwiki/bin/view/XWiki/Foo\">Foo</a>"
             + "</div>", out.toString().trim());
     }
+
+    @Test
+    void getSanitizedURLAttributeValue() throws Exception
+    {
+        HTMLScriptService htmlScriptService = mock(HTMLScriptService.class);
+        this.oldcore.getMocker().registerComponent(ScriptService.class, "html", htmlScriptService);
+        SecurityScriptService securityScriptService = mock(SecurityScriptService.class);
+        this.oldcore.getMocker().registerComponent(ScriptService.class, "security", securityScriptService);
+        URLSecurityScriptService urlSecurityScriptService = mock(URLSecurityScriptService.class);
+        this.oldcore.getMocker().registerComponent(ScriptService.class, "security.url", urlSecurityScriptService);
+        when(securityScriptService.get("url")).thenReturn(urlSecurityScriptService);
+
+        String htmlElement = "input";
+        String attributeName = "value";
```

Comment:
I mean you mock everything here so it doesn't matter, but for the sake of keeping things realistic, I would have used something like `form` and `action` or `a` and `href` as it doesn't make sense to use this macro for the value of an input element.

Reply:
Sure I could change that, here I was thinking about `<input type="button" value="foo"/>` when I wrote the test.

Reply:
Okay, but even for buttons, this is the label that is displayed on the button, so there is absolutely no need to sanitize anything here. The HTML sanitizer explicitly treats `value` as a safe attribute, so it wouldn't return `false` regardless which value you provide.

Reply:
Yes you're right, I was planning to change it after reading your comment anyway